File: java/src/sample/replicant/sample2/MockWidgetRepository.java
// Copyright 2009 Kiel Hodges
package replicant.sample2;

import replicant.*;
import static replicant.Replicant.*;

public class MockWidgetRepository implements WidgetRepository {

  
  public MockWidgetRepository(ExpectationEnforcer expectationEnforcer) {
    findWidget = methodMocker(this, "findWidget").
      mappingResponses().
      enforcingExpectationsWith(expectationEnforcer).
      with1Arg();
    storeWidget = voidMethodMocker(this, "storeWidget").
      enforcingExpectationsWith(expectationEnforcer).
      mappingResponses().
      with1Arg();
  }

  public Widget findWidget(int widgetId) {
    return findWidget.call(widgetId);
  }
  
  public final MethodMocker1<Widget, Integer> findWidget;
  
  
  
  public void storeWidget(Widget widget) {
    storeWidget.call(widget);
  }

  public final MethodMocker1<Void, Widget> storeWidget; 
  
}

File: scala/src/test/scala/sample/MockWidgetRepository.scala
// Copyright 2011 Kiel Hodges
package sample

import replicant._

class MockWidgetRepository extends MockGenericRepository[Widget] { minder =>

  override val mock: WidgetRepository = new BaseSubject with WidgetRepository {
    def findByPartNumber(partNumber: String): Widget = minder.findByPartNumber(partNumber)
  }
    
  val findByPartNumber = method("findByPartNumber", mock.findByPartNumber _)

}
